### PR TITLE
baremetal: add NetworkConfig field

### PR DESCRIPTION
**Pull request by andfasano**
This PR adds a new `install-config` field to store network configuration data (with validation) as described in the enhancement https://github.com/openshift/enhancements/pull/817

### CODE REVIEW

File: pkg/types/baremetal/validation/platform.go
Comment (kirankt) [inline pkg/types/baremetal/validation/platform.go:301]: Suggestion: perhaps we can print the contents of `err` to gain more insight as to why the yaml is invalid?

**Review by kirankt — CHANGES_REQUESTED** [review]

File: pkg/types/baremetal/validation/platform.go
Comment (asalkeld) [inline pkg/types/baremetal/validation/platform.go:15]: shouldn't this be "github.com/ghodss/yaml" ? it looks like most other yaml imports are using ghodss

**Review by asalkeld — COMMENTED** [review]
looks mostly good to me, but not an installer guru..

File: pkg/types/baremetal/validation/platform.go
Comment (andfasano) [reply pkg/types/baremetal/validation/platform.go:15]: Thanks @asalkeld it sounds probably a better choice, especially for validation messages as pointed out by @kirankt

File: pkg/types/baremetal/validation/platform.go
Comment (andfasano) [reply pkg/types/baremetal/validation/platform.go]: Thanks, done! The message is slightly too verbose/technical, but at least could give a better idea about the location of the error

**Review by kirankt — APPROVED** [review]
Looks good to me, but lets wait for the enhancement to merge before committing this.
/hold

**Comment by kirankt** [comment]
/approve
Please remove the hold once the enhancement merges

**Comment by imain** [comment]
LGTM

**Comment by romfreiman** [comment]
> Do you have a reference to a specific failed mandatory build?

Why failed? :) I dont know, that's why I suggested to verify it prior to pushing

**Review by kirankt — APPROVED** [review]
/approve

File: pkg/types/baremetal/validation/platform.go
Comment (staebler) [inline pkg/types/baremetal/validation/platform.go:301]:
```suggestion
			errors = append(errors, field.Invalid(fldPath.Index(idx).Child("networkConfig"), host.NetworkConfig, fmt.Sprintf("Not valid yaml: %s", err.Error())))
```

File: pkg/types/baremetal/platform.go
Comment (staebler) [inline pkg/types/baremetal/platform.go:37]: Please add documentation here for this field so that it shows up in the `openshift-install explain` output.

Comment (staebler) [inline pkg/types/baremetal/platform.go:37]: Are we sure that we want this to be a `string` rather than a `map[string]interface{}`? I would expect the user to want to enter in-line yaml rather than a text string.

**Review by staebler — COMMENTED** [review]
/approve

File: pkg/types/baremetal/platform.go
Comment (andfasano) [reply pkg/types/baremetal/platform.go:37]: The installer manages it essentially as a blob (it will be stored also in a secret, see PR [5247](https://github.com/openshift/installer/pull/5247)) and will pass it as it is to the image builder container (with a just a minimal validation).

**Pull request merged**
